hrjobcontract/js/src/job-contract/controllers/modal/modal-contract-new.controller.js

```diff
@@ -285,6 +290,105 @@ define([
         $scope.entity.leave[index].add_public_holidays = utils.absenceTypes[absenceTypeID].add_public_holiday_to_entitlement;
       });
     }
+
+    /**
+     * Opens option editor window for contract type, location, end reason or insurance
+     *
+     * @param optionUrl
+     * @param fieldName
+     */
+    function openOptionsEditor (optionUrl, fieldName) {
+      var optionTypes = {
+        'hrjobcontract_details_contract_type': 'contract_type',
+        'hrjobcontract_details_location': 'location',
+        'hrjobcontract_details_end_reason': 'end_reason',
+        'hrjobcontract_health_health_plan_type': 'provider_life_insurance'
+      };
+
+      crmAngService.loadForm(optionUrl)
+        .on('crmUnload', function () {
+          if (fieldName === 'hrjobcontract_health_health_plan_type') {
+            contractHealthService.getOptions(fieldName, true)
+              .then(function (data) {
+                var healthOptions = {};
```

Comment:
Please use Lodash here:

```js
_.mapValues(_.indexBy([{key: 1, value: 'one'},{key: 2, value: 'two'}], 'key'), 'value')
>>> {1: "one", 2: "two"}
```

So it would be smth like:

```js
var healthOptions = _.mapValues(_.indexBy(data, 'key'), 'value');
```

Also, between `var x = ...` and other types of expressions, like function calls, please leave an empty line break.